addons/html_editor/static/src/backend/dynamic_field/dynamic_field_plugin.js

```diff
@@ -0,0 +1,371 @@
+    "font",
+    "font-size",
+    "font-family",
+    "link",
+];
+
+export class DynamicFieldPlugin extends Plugin {
+    static id = "dynamicField";
+    static dependencies = ["selection", "history", "overlay", "dom", "toolbar", QWebPlugin.id];
+
+    /** @type {import("plugins").EditorResources} */
+    resources = {
+        unsplittable_node_predicates: (node) =>
+            node.nodeType === Node.ELEMENT_NODE && node.matches(".page, .header, .footer"),
+        toolbar_groups: withSequence(9, { id: "dynamic_field" }),
+        toolbar_items: [
+            {
+                id: "editDynamicField",
+                groupId: "dynamic_field",
+                namespaces: ["compact", "expanded"],
+                commandId: "editDynamicField",
+            },
+        ],
+        user_commands: [
+            {
+                id: "insertField",
+                title: _t("Field"),
+                description: _t("Insert a field"),
+                icon: "fa-database",
+                run: this.insertField.bind(this),
+                isAvailable: (selection) => isHtmlContentSupported(selection),
+            },
+            {
+                id: "editDynamicField",
+                title: _t("Edit field"),
+                description: _t("Change the placeholder or the expression for an existing field"),
+                icon: "fa-pencil",
+                run: this.editField.bind(this),
+                isAvailable: () => !!this.getPopoverTarget(true),
+            },
+        ],
+        powerbox_categories: withSequence(1, {
+            id: "dynamic_field_tools",
+            name: _t("Data"),
+        }),
+        powerbox_items: [
+            withSequence(20, {
+                categoryId: "dynamic_field_tools",
+                commandId: "insertField",
+            }),
+        ],
+        selectionchange_handlers: withSequence(9, this.onSelectionChanged.bind(this)),
+        dynamic_model_change_handlers: this.updateDynamicModel.bind(this),
+        toolbar_buttons_disabled_predicates: this.isButtonDisabled.bind(this),
+        normalize_handlers: withSequence(11, this.normalizeQwebPlaceholders.bind(this)),
+        clipboard_content_processors: withSequence(11, this.cleanQwebExpressionsForCopy.bind(this)),
+        clean_for_save_handlers: withSequence(11, this.cleanQwebExpressionsForSave.bind(this)),
+    };
+
+    fieldTagName = "T";
+    fieldAttribute = "t-out";
+
+    setup() {
+        this.resModel = this.config.dynamicResModel;
+
+        this.fieldPopover = this.dependencies.overlay.createOverlay(FieldSelectorPopover, {
+            hasAutofocus: true,
+            editable: this.editable,
+            className: "bg-light",
+        });
+
+        this.addDomListener(
+            this.editable,
+            "click",
+            (ev) => {
+                if (ev.detail === 1) {
+                    this.setFullSelection(ev.srcElement);
+                }
+            },
+            true
+        );
```

Comment:
This reads like if there is a click, any click, then this code will "set full selection" of the element. This is misleading because it's not what it does. IMO it would be cleaner if the code was:
```js
this.addDomListener(
    this.editable,
    "click",
    (ev) => {
        if (ev.detail === 1 && this.isMySpecialElement(ev.srcElement)) {
            this.setFullSelection(ev.srcElement);
        }
    },
    true
);
```
So that when someone calls `setFullSelection` it really _does_ set a "full selection". It makes more sense to me at least.